### ISSUE TITLE
Using arrow functions

### ISSUE THREAD

**Issue by torayeff** [Urigo/angular-meteor#965]
Hi @Urigo. This is not real issue but a misunderstanding. I am following angular-meteor tutorial, and in some places you use arrow functions syntax and in some places normal functions syntax. For example:
```
this.getUserById = (userId) => {
        return Meteor.users.findOne(userId);
      };
```
or

```
Meteor.methods({
  invite: function (partyId, userId) {
    check(partyId, String);
    check(userId, String);
...
```

If I use arrow function syntax in Meteor.methods or in directive constructor it will not work. Can you please explain when and where I can use arrow functions syntax?

**Comment by mgrivera** [comment]
Hi, 
Maybe I'm wrong, but I think you can use arrow function wherever you want. In the end, they will all be transpiled by babel to normal functions and will be all the same in that respect ...

**Comment by Tallyb** [comment]
Almost. Arrow functions do not change your "this"

**Comment by jacobdr** [comment]
*tldr;* arrow functions are awesome because of their conciseness (syntactic
sugar) and ability to bind enclosing 'this' in a way that makes sense for a
majority of use cases, but can also lead to issues with libraries that
manipulate context regularly in their course of business

To expand on Tally's answer a little bit more, arrow functions
<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions>
keep bounded the lexical scope of the new arrow function's calling context.
WHEW a mouthful.

Put another way, the "this" context of the parent and all the values
attached to 'this' are available in the child via the same "this".

Normally, whenever a  new "anonymous" (and unbounded to an object) function
is created -- e.g. `function test(){this.myValue = 10; return
this.myValue}` -- a new "this" is instantiated every time. A new 'this' for
every use of the keyword 'function'.

Think of an arrow function as never creating a new "this" -- the "this"
from the function's creation context is placed inside the body of the arrow
function.

What makes the initial question interesting is that for years Javascript
library authors have been using fancy language semantics like `bind` and
`call` to create beautiful APIs. Think of the method chaining of jQuery as
an example of this. Another example are the Meteor authors who attach
useful objects into the context of various functions -- such as inside
method calls, or inside subscribe calls, etc., where we can access things
like 'this.user'.

Arrow functions -- because they alter the traditional functional Javascript
lexical scoping -- can in some instances override the intent and
implementations of library authors, and lead to unexpected behaviors.

**Comment by Urigo** [comment]
@jacobdr great answer, I've also linked to it from the [SO question](http://stackoverflow.com/a/34366429/1426570)

**Comment by torayeff** [comment]
@Urigo @jacobdr Thanks for short and simple explanations.
